### PR TITLE
Words and test results routers and services

### PR DESCRIPTION
**Pull request by lebedevaanastasiya**

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -6,5 +6,8 @@ router.use(authenticate());
 
 router.use('/auth', require('./auth'));
 router.use('/users', require('./users'));
+router.use('/words', require('./words'));
+router.use('/word_topics', require('./wordTopics'));
```
Comment (mbarouski) [inline routes/index.js]: camelCasePlease

File: routes/testResults/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   testResultsService.getAllByUserIdTopicId(req.params.userId, req.params.topicId)
+       .then(data => {
+           res.json({
+               data: data
+           });
+       });
+});
+
+router.post('/', (req, res, next) => {
+   testResultsService.create(req.body.testResult)
```
Comment (mbarouski) [inline routes/testResults/index.js]: `req.body`

File: routes/wordTopics/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    wordTopicsService.getAll()
+        .then(data => {
+            res.json({
+                data: data
+            });
+        });
+});
+
+router.post('/', (req, res, next) => {
+   wordTopicsService.create({name: req.body.name})
```
Comment (mbarouski) [inline routes/wordTopics/index.js]: `req.body`

**Review by mbarouski — CHANGES_REQUESTED** [review]

File: routes/index.js [hunk 4]
```diff
@@ -2,14 +2,14 @@ const express = require('express');
 const router = express.Router();
 const authenticate = require('../services/auth').authenticate;
 
-router.use(authenticate());
+//router.use(authenticate());
```
Comment (mbarouski) [inline routes/index.js]: uncomment, please

File: routes/testResults/index.js [hunk 5]
```diff
@@ -12,8 +12,8 @@ router.get('/:userId', (req, res, next) => {
         });
 });
 
-router.get('/:userId/:topicId', (req, res, next) => {
-   testResultsService.getAllByUserIdTopicId(req.params.userId, req.params.topicId)
+router.get('/:userId/:testId', (req, res, next) => {
+   testResultsService.getAllByUserIdTestId(req.params.userId, req.params.testId)
```
Comment (mbarouski) [inline routes/testResults/index.js]: how we can get all results of user?

File: services/testResults/index.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+            })
+    })
+}
+
+function getAllByUserId(userId) {
+    return TestResult.findAll({where: {userId: userId}});
+}
+
+function getAllByUserIdTestId(userId, testId) {
+    return TestResult.findAll({where: {userId: userId, testId: testId}});
```
Comment (mbarouski) [inline services/testResults/index.js:20]: what does it mean... we can have only one result with userId and testId

File: services/wordTopics/index.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+const WordTopic = require('../../database/models/wordTopics');
+
+function getAll() {
+    return WordTopic.findAll();
+}
+
+function create(wordTopic) {
+    return new Promise((resolve, reject) => {
+        WordTopic.findOrCreate({where: {name: wordTopic.name}, default: {name: wordTopic.name}})
```
Comment (mbarouski) [inline services/wordTopics/index.js]: if you use field in `where` you can not use it in defaults

Comment (mbarouski) [reply services/wordTopics/index.js]: `default` => `defaults`

**Review by mbarouski — CHANGES_REQUESTED** [review]

File: routes/testResults/index.js (hunk 5 above)
Comment (lebedevaanastasiya) [reply routes/testResults/index.js]: There is getAllByUserId() function

File: services/testResults/index.js (hunk 6 above)
Comment (lebedevaanastasiya) [reply services/testResults/index.js:20]: User can do some test several times

Comment (lebedevaanastasiya) [reply services/testResults/index.js:20]: And users results by passing this test can be different

**Pull request merged**